From: Emmanuele Bassi <ebassi@gnome.org>
Date: Mon, 10 Sep 2018 12:44:59 +0100
Subject: Install m4 macros unconditionally

Partial revert of commit a7a6449f4d63357bf3a66140e9fb3ccb18611c0b.

Checking for the availability of m4 for installing m4 macro files
creates an implicit dependency on m4 even if GLib does not need it; this
prevents building GLib and then installing Autotools in order to build a
project that depends on GLib.

Bug: https://gitlab.gnome.org/GNOME/glib/issues/1520
Origin: upstream, 2.59.0, commit:e8fde0b9e7d9c8f76b374f67188f0229c6b01e42
---
 meson.build | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/meson.build b/meson.build
index a6b372b..09e12aa 100644
--- a/meson.build
+++ b/meson.build
@@ -1878,7 +1878,6 @@ python_name = 'python3'
 
 # Determine which user environment-dependent files that we want to install
 have_bash = find_program('bash', required : false).found() # For completion scripts
-have_m4 = find_program('m4', required : false).found() # For m4 macros
 have_sh = find_program('sh', required : false).found() # For glib-gettextize
 
 # FIXME: How to detect Solaris? https://github.com/mesonbuild/meson/issues/1578
@@ -1990,11 +1989,9 @@ if have_sh
     configuration : gettextize_conf)
 endif
 
-if have_m4
-  # Install m4 macros that other projects use
-  install_data('m4macros/glib-2.0.m4', 'm4macros/glib-gettext.m4', 'm4macros/gsettings.m4',
-    install_dir : join_paths(get_option('datadir'), 'aclocal'))
-endif
+# Install m4 macros that other projects use
+install_data('m4macros/glib-2.0.m4', 'm4macros/glib-gettext.m4', 'm4macros/gsettings.m4',
+  install_dir : join_paths(get_option('datadir'), 'aclocal'))
 
 if host_system != 'windows'
   # Install Valgrind suppression file (except on Windows,
